ocr-numbers.json

```diff
@@ -0,0 +1,187 @@
+          "   "
+        ],
+        "expected": "?"
+      },
+      {
+        "description": "Input with a number of lines that is not a multiple of four raises an error",
+        "input": [
+          " _ ",
+          "| |",
+          "   "
+        ],
+        "expected": -1
+      },
+      {
+        "description": "Input with a number of columns that is not a multiple of three raises an error",
+        "input": [
+          "    ",
+          "   |",
+          "   |",
+          "    "
+        ],
+        "expected": -1
+      },
+      {
+        "description": "Recognizes 110101100",
+        "input": [
+          "       _     _        _  _ ",
+          "  |  || |  || |  |  || || |",
+          "  |  ||_|  ||_|  |  ||_||_|",
+          "                           "
+        ],
+        "expected": "110101100"
+      },
+      {
+        "description": "Garbled numbers in a string are replaced with ?",
+        "input": [
+          "       _     _           _ ",
+          "  |  || |  || |     || || |",
+          "  |  | _|  ||_|  |  ||_||_|",
+          "                           "
+        ],
+        "expected": "11?10?1?0"
+      },
+      {
+        "description": "Recognizes 2",
+        "input": [
+          " _ ",
+          " _|",
+          "|_ ",
+          "   "
+      ],
+      "expected": "2"
+      },
+      {
+      "description": "Recognizes 3",
+      "input": [
+        " _ ",
+        " _|",
+        " _|",
+        "   "
+      ],
+      "expected": "3"
+      },
+      {
+      "description": "Recognizes 4",
+      "input": [
+        "   ",
+        "|_|",
+        "  |",
+        "   "
+      ],
+      "expected": "4"
+      },
+      {
+      "description": "Recognizes 5",
+      "input": [
+        " _ ",
+        "|_ ",
+        " _|",
+        "   "
+      ],
+      "expected": "5"
+      },
+      {
+      "description": "Recognizes 6",
+      "input": [
+        " _ ",
+        "|_ ",
+        "|_|",
+        "   "
+      ],
+      "expected": "6"
+      },
+      {
+      "description": "Recognizes 7",
+      "input": [
+        " _ ",
+        "  |",
+        "  |",
+        "   "
+      ],
+      "expected": "7"
+      },
+      {
+      "description": "Recognizes 8",
+      "input": [
+        " _ ",
+        "|_|",
+        "|_|",
+        "   "
+      ],
+      "expected": "8"
+      },
+      {
+      "description": "Recognizes 9",
+      "input": [
+        " _ ",
+        "|_|",
+        " _|",
+        "   "
+      ],
+      "expected": "9"
+      },
+      {
+      "description": "Recognizes string of decimal numbers",
+      "input": [
+        "    _  _     _  _  _  _  _  _ ",
+        "  | _| _||_||_ |_   ||_||_|| |",
+        "  ||_  _|  | _||_|  ||_| _||_|",
+        "                              "
+      ],
+      "expected": "1234567890"
+      },
+      {
+      "description": "Numbers separated by empty lines are recognized. Empty lines are replaced with commas",
```

Comment:
well, it's not really that there's any additional separation, since the number of lines is still a multiple of 4, and the fourth line has always been empty. the trailing empty line was not replaced with a comma either. I may have just said "Recognizes multiple lines of numbers. The lines are joined by commas."